### ISSUE TITLE
Multiple github files

### ISSUE THREAD

**Issue by sakulstra** [tinacms/tinacms#1284]
Disclaimer: Not sure if that is a feature request/docs request or question. The solution proposed here seems to work, so worst case we can just close it and with some luck other people struggling with this will find it :sweat: 

~actually the solution proposed here doesn't fully work yet as you cannot save changes - have to still investigate why that is~
Replacing `usePlugins([configForm, form]);` with `usePlugin(form); usePlugin(configForm);` solved the issue :man_shrugging: 

## Summary
I'd like to configure my pages with basically two files - one with "global settings", which is used by all pages and one that extends or overwrites them. I found a way to do this (in the basic example section), but I was wondering if `getGithubPreviewProps` should be able to handle multiple files at once instead :thinking: 

### Basic example
Let's imagine you allow configuring meta title & description per page. Somewhere you'd like an also configurable fallback meta config.

Currently i solved this ~ like that
```js
// is used like "export const getStaticProps = getTinaProps("home");"
export const getTinaProps = (file: string) => async ({
  preview,
  previewData,
}) => {
  if (preview) {
    const contentFile = await getGithubPreviewProps({
      ...previewData,
      fileRelativePath: `json/${file}.json`,
      parse: parseJson,
      head_branch: "tina",
    });
    return {
      props: {
        ...contentFile.props,
        globalFile: (
          await getGithubPreviewProps({
            ...previewData,
            fileRelativePath: `json/global.json`,
            parse: parseJson,
            head_branch: "tina",
          })
        ).props.file,
      },
    };
  }
  return {
    props: {
      sourceProvider: null,
      error: null,
      preview: false,
      globalFile: {
        fileRelativePath: `content/global.json`,
        data: (await import(`../../../json/global.json`))
          .default,
      },
      file: {
        fileRelativePath: `content/${file}.json`,
        data: (await import(`../../../json/${file}.json`))
          .default,
      },
    },
  };
};
```
to essentially load two files.

Then on the client i create two githubJsonForm plugins and merge the page:
```js
export function useCustomFiles({ file, globalFile }) {
  const [page, form] = useGithubJsonForm(file, meta);
  const [configPage, configForm] = useGithubJsonForm(globalFile, globalConfig);

  usePlugin(form);
  usePlugin(configForm);

  // Register it with the CMS
  useGithubToolbarPlugins();
  return { page: { ...configPage, ...page }, form }; // probably better deep merged
}
```
### Motivation
I'd like to omit duplicating config across pages to keep them in sync.

**Comment by ncphillips** [comment]
Love it. We've definitely felt this same pain on tinacms.org. Thanks @sakulstra

**Comment by ncphillips** [comment]
Hey! We added this [getGithubFile](https://github.com/tinacms/tinacms/tree/master/packages/next-tinacms-github#_getgithubfile) function which should make this a bit less clunky!

**Comment by ncphillips** [comment]
Total aside that global content default approach is really interesting!

**Comment by ncphillips** [comment]
Hey we added a [getGithubFile](https://github.com/tinacms/tinacms/tree/master/packages/next-tinacms-github#_getgithubfile) function to `next-tinacms-github`. Does this function look like it would solve your problem?

**Comment by ncphillips** [comment]
As an aside, that fallback config idea is really interesting. You might benefit from using [`useFormScreenPlugin(globalConfig)`](https://tinacms.org/docs/plugins/screens#useformscreenplugin)!

**Comment by sakulstra** [comment]
@ncphillips I tried `getGithubFile` but didn't manage to migrate due to "cannot open file or sth similar" (didn't have time to debug yet, will do so and create a pr/issue if appropriate)

`useFormScreenPlugin` was a great suggestion and improved the story a lot! :pray: :+1:

---

The main pain point though for me is merging different page values and accessing them in deeply nested components.

For now I solved that by using a custom global context which i initialize with a deep merge of multiple loaded files but that's kinda weird - wonder if there could be a better built in solution like `useTinaContext()` which would contain the pages.
